### ISSUE TITLE
%pfile doesn't detect objects from the standard library

### ISSUE THREAD

**Issue by illright** [ipython/ipython#12960]
## Steps to reproduce

* Execute `import math`
* Execute `%pfile math`
* Execute `%pfile math.log`

## Expected result
Both calls to `%pfile` tell the user that `math` is a part of the standard library

## Actual result
Both calls to `%pfile` produce an error:
```
No file found for math
No file found for math.log
```

## System info
Manjaro Linux, Python 3.9.4, IPython 7.22.0 (reproduces on IPython 8.0.0.dev)

**Comment by illright** [comment]
I think it could be helpful to point users towards the standard library in such cases, regardless of their setup.

Also, I don't really understand the mechanics of `%pfile`. I created a file `test.py` in the current directory with the contents `specific_var = 5`, then executed `from test import specific_var` and then `%pfile specific_var`. I expected it to point me towards `file.py`, but it also printed out `No file found for specific_var`.

**Comment by MrMino** [comment]
I disagree. It would be very difficult to tell whether something is _actually_ stdlib or not, you'd have to maintain a list throughout each platform × python version, which would be a nightmare'ish endeavour.

>  I expected it to point me towards file.py, but it also printed out No file found for specific_var.

As I understand it, you're giving it the value `5`, which is not defined anywhere where `%pfile` can look. The docs say:

> Print (or run through pager) the source code for an object.

`5` is not defined in your file, and you cannot get its source code by using Python's introspection tools, hence `No file found for specific_var`.
